index.bs

```diff
@@ -511,6 +562,53 @@ The <dfn method for=FileSystemFileHandle>createWritable(|options|)</dfn> method,
 
 </div>
 
+### The {{FileSystemFileHandle/createSyncAccessHandle()}} method ### {#api-filesystemfilehandle-createsyncaccesshandle}
+
+<div class="note domintro">
+  : |handle| = await |fileHandle| . {{FileSystemFileHandle/createSyncAccessHandle()|createSyncAccessHandle}}({ {{FileSystemFileHandleCreateAccessHandleOptions/mode}}: "in-place" })
+  :: Returns a {{FileSystemSyncAccessHandle}} that can be used to read/write from/to the file.
+     Changes made through |handle| might be immediately reflected in the file represented by |fileHandle|.
+     To ensure the changes are reflected in this file, the handle must be flushed or closed.
+
+    Creating a {{FileSystemSyncAccessHandle}} [=file entry/lock/take|takes an exclusive lock=] on the
+    [=FileSystemHandle/entry=] associated with |fileHandle|. This prevents the creation of
+    further {{FileSystemSyncAccessHandle|FileSystemSyncAccessHandles}} or {{FileSystemWritableFileStream|FileSystemWritableFileStreams}}
+    for the entry, until the access handle is closed.
+
+    The returned {{FileSystemSyncAccessHandle}} offers synchronous {{FileSystemFileHandle/read()}} and
+    {{FileSystemFileHandle/write()}} methods. This allows for higher performance for critical methods on
+    contexts where asynchronous operations come with high overhead, i.e., WebAssembly.
+
+    For the time being, this method will only succeed when the |fileHandle| belongs to the
+    [=origin private file system=]. Also temporarily, the {{FileSystemFileHandleCreateAccessHandleOptions/mode}} parameter
+    must be set to "in-place". This prevents us from setting a default behavior and allows us to bring access handles to
+    other filesytems in the future.
+
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemFileHandle>createSyncAccessHandle(|options|)</dfn> method, when invoked, must run these steps:
+
+1. Let |result| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |permissionStatus| be the result of [=requesting file system permission=]
+     given <b>[=this=]</b> and {{"readwrite"}}.
+     If that throws an exception, [=reject=] |result| with that exception and abort.
+  1. If |permissionStatus| is not {{PermissionState/"granted"}},
+     reject |result| with a {{NotAllowedError}} and abort.
+  1. Let |entry| be <b>[=this=]</b>'s [=FileSystemHandle/entry=].
+  1. If |entry| does not represent an [=/entry=] in an [=origin private file system=],
```

Comment:
We currently don't have tests to check this exception order. Since NotAllowedError would fit as well (it's a not recoverable error, and it is a limitation being imposed by the platform), and to avoid adding potentially combinatorial tests on exception order, I would change this error to NotAllowedError. WDYT?

Reply:
That seems reasonable, although it removes the ability for web author code to distinguish these cases and do something about it. You could imagine a bug in a web app where they are accidentally using this API on non-OPFS files, and they have

```js
try {
  useAPIIncorrectly();
} catch (e) {
  if (e.name === "NotAllowedError") {
    tellTheUser("Please grant the permission!!");
  }
}
```

which would result in a more-confusing user experience.

Reply:
I migrated this comment to https://github.com/whatwg/fs/pull/21/files#r848524692 in order to close this PR!